numba/tests/test_entrypoints.py

```diff
@@ -0,0 +1,51 @@
+import sys
+import types
+
+import pkg_resources
+
+from .support import TestCase
+
+
+class TestEntrypoints(TestCase):
+    """
+    Test registration of init() functions from Numba extensions
+    """
+
+    def test_init_entrypoint(self):
+        # loosely based on Pandas test from:
+        #   https://github.com/pandas-dev/pandas/pull/27488
+
+        # FIXME: Python 2 workaround because nonlocal doesn't exist
+        counters = {'init': 0}
+
+        def init_function():
+            counters['init'] += 1
+
+        mod = types.ModuleType("_test_numba_extension")
+        mod.init_func = init_function
+        sys.modules[mod.__name__] = mod
```

Comment:
I'm not sure that mutating `sys.modules` in the main process is a good idea for a reproducible and consistent test suite, perhaps run all this in a child process? (doing this would also resolve the global state problem below I think).

Reply:
Is there an existing mechanism to run this test in a child process?  I'm worried about slowing down the test suite even further when a try/finally should be sufficient to restore state.

Reply:
Given the recently added and similar https://github.com/numba/numba/blob/master/numba/tests/test_jit_module.py#L66-L81, I think this is perhaps ok until proven otherwise.